manager/orchestrator/global/global.go

```diff
@@ -171,6 +180,58 @@ func (g *Orchestrator) Run(ctx context.Context) error {
 		case <-g.stopChan:
 			return nil
 		}
+		g.tickTasks(ctx)
+	}
+}
+
+// FixTask validates a task with the current cluster settings, and takes
+// action to make it conformant to node state and service constraint
+// it's called at orchestrator initialization
+func (g *Orchestrator) FixTask(ctx context.Context, batch *store.Batch, t *api.Task) {
+	service, exists := g.globalServices[t.ServiceID]
+	if !exists {
+		return
+	}
+	// if a task's DesiredState has past running, the task has been processed
+	if t.DesiredState > api.TaskStateRunning {
+		return
+	}
+
+	var node *api.Node
+	if t.NodeID != "" {
+		node = g.nodes[t.NodeID]
+	}
+	// if the node no longer valid, remove the task
+	if t.NodeID == "" || orchestrator.InvalidNode(node) {
+		g.removeTask(ctx, batch, t)
+		return
+	}
+	// if the node doesn't meet service constraint, remove the task
```

Comment:
This is redundant with `constraintenforcer`. Either the orchestrator should not do this at init, or `constraintenforcer` should not do it. I'd vote for keeping it in `constraintenforcer` because that seems like a logical place, and it's less of a change that way.